### ISSUE TITLE
Handling migrated datasets

### ISSUE THREAD

**Issue by MikeBauerCA** [zowe/zowe-cli#953]
@zszabo14 commented on [Tue Feb 02 2021](https://github.com/zowe/community/issues/1014)

**Hi ZOWE CLI Team**

I just had a strange error when eg. copied a directory to a PDS. :
`Command Error:
Upload a directory with multiple files to a physical sequential data set is not permitted`

and for some others I saw:

```
`success: false
from:  /home/jenkins/workspace/ckage-ZTF1-DEV_WWFD-redeployment/UAT/WFR00001/BATL/WFTGCSX0
to:  WWFT.WFR00001.BATL
error: 
 message: 
 z/OSMF REST API Error:
 Rest API failure with HTTP(S) status 500
 category: 6
 rc: 8
 reason: 513
 message: fopen() error
 details: 
 - EDC5047I An invalid file name was specified as a function parameter. (errno2=0xC00B0283)
 - __amrc: __last_op=401 __code=0x00000000
 
 stack: 
 - Error: z/OSMF REST API Error:
 - Rest API failure with HTTP(S) status 500
 - category: 6
 - rc: 8
 - reason: 513
 - message: fopen() error
 - details: 
 -  - EDC5047I An invalid file name was specified as a function parameter. (errno2=0xC00B0283)
 -  - __amrc: __last_op=401 __code=0x00000000
 -  at ZosmfRestClient.populateError (/usr/lib/npm-global/lib/node_modules/@zowe/cli/node_modules/@zowe/imperative/lib/rest/src/client/AbstractRestClient.js:507:16)
 -  at ZosmfRestClient.onEnd (/usr/lib/npm-global/lib/node_modules/@zowe/cli/node_modules/@zowe/imperative/lib/rest/src/client/AbstractRestClient.js:427:31)
 -  at IncomingMessage.<anonymous> (/usr/lib/npm-global/lib/node_modules/@zowe/cli/node_modules/@zowe/imperative/lib/rest/src/client/AbstractRestClient.js:362:18)
 -  at IncomingMessage.emit (events.js:326:22)
 -  at endReadableNT (_stream_readable.js:1241:12)
 -  at processTicksAndRejections (internal/process/task_queues.js:84:21)

Command Error:
z/OSMF REST API Error:
Rest API failure with HTTP(S) status 500
category: 6
rc: 8
reason: 513
message: fopen() error
details: 
 - EDC5047I An invalid file name was specified as a function parameter. (errno2=0xC00B0283)
 - __amrc: __last_op=401 __code=0x00000000
```

It turned out, the PDS were migrated. These errors are misleading.

**Please add a check to each command (eg submit job or any other) to check if the refered dataset is migrated and an option like: --recall . 
Also, would be nice to have a command for recall: `zowe zos-files recall <dataset(s)> --wait` (wait or not until recall done)**

**Thank You!**

---

@balhar-jakub commented on [Wed Feb 17 2021](https://github.com/zowe/community/issues/1014#issuecomment-780410462)

@MikeBauerCA Doesn't this look like something related to the CLI?

**Comment by MikeBauerCA** [comment]
@zszabo14, there is a `zowe files recall ds <dataSetName>` command that accepts a `--wait` flag. Is your second request to allow for multiple datasets to be recalled in a single command?

**Comment by gejohnston** [comment]
This enhancement has had no activity for 12 months. The issue also has less than 5 up-votes by the community. No action on this enhancement is targeted for the next 2 calendar quarters. Therefore, this enhancement is being closed. If you feel that this enhancement should continue to be available for community up-votes, you may reopen this issue.
